client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -242,13 +252,17 @@ const Checkout = React.createClass( {
 			} );
 		}
 
+		if ( ! selectedSiteSlug ) {
```

Comment:
Maybe fall back to `/checkout/thank-you/features` since `/checkout/thank-you` still takes us to the checkout page with infinite loaders.